src/components/item/settings/ThumbnailSetting.tsx

```diff
@@ -96,10 +101,10 @@ const ThumbnailSetting = ({ item }: Props): JSX.Element | null => {
   };
 
   const alt = translateBuilder(BUILDER.THUMBNAIL_SETTING_MY_THUMBNAIL_ALT);
-  const imgUrl = item.settings?.hasThumbnail
-    ? thumbnailUrl
-    : (item.type === ItemType.LINK ? item.extra[ItemType.LINK] : {})
-        ?.icons?.[0];
+  let imgUrl = item.settings?.hasThumbnail ? thumbnailUrl : null;
+  if (!imgUrl && item.type === ItemType.LINK) {
+    imgUrl = getLinkThumbnailUrl(item.extra);
+  }
```

Comment:
Maybe you can use a function for that ?

```suggestion
const getImgUrl = (item, thumbnailUrl) => {
  if (item.settings?.hasThumbnail) {
    return thumbnailUrl;
  } else if (!imgUrl && item.type === ItemType.LINK) {
    return getLinkThumbnailUrl(item.extra);
  } else {
    return null;
  }
}

...

const imgUrl = getImgUrl();
```

Reply:
I agree with your suggestion, I think anyway this code is a duplicata and should maybe be handled at the query client level (we always check thumbnail before we fallback to icon/thumbnail), I would even prefer save those thumbnails as normal thumbnail so we don't have problem checking for both each time. But it's future story..